client/client.go

```diff
@@ -844,14 +844,14 @@ func (r *NotaryRepository) saveMetadata(ignoreSnapshot bool) error {
 	if err != nil {
 		return err
 	}
-	err = r.cache.Set(data.CanonicalRootRole, rootJSON)
+	err = r.cache.Set(data.CanonicalRootRole.String(), rootJSON)
 	if err != nil {
 		return err
 	}
 
-	targetsToSave := make(map[string][]byte)
+	targetsToSave := make(map[data.RoleName][]byte)
 	for t := range r.tufRepo.Targets {
-		signedTargets, err := r.tufRepo.SignTargets(t, data.DefaultExpires(data.CanonicalTargetsRole))
+		signedTargets, err := r.tufRepo.SignTargets(data.RoleName(t), data.DefaultExpires(data.CanonicalTargetsRole))
```

Comment:
`tufRepo.Targets` is already a `map[data.RoleName]*data.SignedTargets`, so this cast may not be necessary.